homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -361,6 +416,25 @@ async def async_setup_entry(
             EnvoyConsumptionEntity(coordinator, description)
             for description in CONSUMPTION_SENSORS
         )
+    # For each production phase reported add production entities
+    if envoy_data.system_production_phases:
+        for use_phase in envoy_data.system_production_phases:
+            if envoy_data.system_production_phases[use_phase] is not None:
+                entities.extend(
+                    EnvoyProductionPhaseEntity(coordinator, description)
+                    for description in PRODUCTION_PHASE_SENSORS
+                    if description.on_phase == use_phase
+                )
```

Comment:
```suggestion
        entities.extend(
            EnvoyProductionPhaseEntity(coordinator, description)
            for description in PRODUCTION_PHASE_SENSORS
            if description.on_phase == use_phase
            for use_phase, phase in envoy_data.system_production_phases.items()
            if phase is not None
        )
```

Reply:
But we can avoid iterating `PRODUCTION_PHASE_SENSORS` if we change it to a dict https://github.com/home-assistant/core/pull/108725/files#r1468660061

Reply:
Using PRODUCTION_PHASE_SENSORS as dict (above) condensed this to:
```py
        entities.extend(
            EnvoyProductionPhaseEntity(coordinator, description)
            for use_phase, phase in envoy_data.system_production_phases.items()
            for description in PRODUCTION_PHASE_SENSORS[use_phase]
            if phase is not None
        )
```

Reply:
tested but not committed yet. If ok I'll commit.

Reply:
Looks much better. Go for it